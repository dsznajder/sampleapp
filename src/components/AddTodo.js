import React, { Component } from 'react'

export default class TodoAdd extends Component {
  state = {}

  updateValue = ({ target: { value } }) => this.setState({ value })

  addTodo = () => {
    const { value } = this.state
    value && this.props.addTodo(value)
    this.setState({ value: '' })
  }

  onEnter = ({ key }) => key === 'Enter' && this.addTodo()

  render() {
    return (
      <div>
        <input onChange={this.updateValue} onKeyDown={this.onEnter} value={this.state.value} />
        <button disabled={!this.state.value} onClick={this.addTodo}>
          Add
        </button>
      </div>
    )
  }
}
