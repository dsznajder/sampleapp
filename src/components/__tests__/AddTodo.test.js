import React from 'react'
import renderer from 'react-test-renderer'

import AddTodo from '../AddTodo'

describe('<AddTodo />', () => {
  const defaultProps = {
    addTodo: jest.fn(),
  }

  const wrapper = renderer.create(<AddTodo {...defaultProps} />)

  test('render', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
