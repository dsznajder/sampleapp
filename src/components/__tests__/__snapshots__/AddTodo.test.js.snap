// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTodo /> render 1`] = `
<div>
  <input
    onChange={[Function]}
    onKeyDown={[Function]}
    value={undefined}
  />
  <button
    disabled={true}
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;
